packages/adblocker-webextension/adblocker.ts

```diff
@@ -505,6 +614,54 @@ export class WebExtensionBlocker extends FiltersEngine {
           },
     );
   }
+
+  private executeScriptlets(browser: Browser, tabId: number, scripts: string[]): void {
+    // Dynamically injected scripts scripts can be difficult to find later in
+    // the debugger. Console logs simplifies setting up breakpoints if needed.
+    let debugMarker;
+    if (this.config.debug) {
+      debugMarker = (text: string) =>
+        `console.log('[ADBLOCKER-DEBUG]:', ${JSON.stringify(text)});`;
+    } else {
+      debugMarker = () => '';
+    }
+
+    // the scriptlet code that contains patches for the website
+    const codeRunningInPage = `(function(){
+${debugMarker('run scriptlets (executing in "page world")')}
+${scripts.join('\n\n')}}
+)()`;
+
+    // wrapper to break the "isolated world" so that the patching operates
+    // on the website, not on the content script's isolated environment.
+    const codeRunningInContentScript = `
+(function(code) {
+    ${debugMarker('run injection wrapper (executing in "content script world")')}
+    var script;
+    try {
+      script = document.createElement('script');
+      script.appendChild(document.createTextNode(decodeURIComponent(code)));
+      (document.head || document.documentElement).appendChild(script);
+    } catch (ex) {
+      console.error('Failed to run script', ex);
+    }
+    if (script) {
+        if (script.parentNode) {
+          script.parentNode.removeChild(script);
+        }
+        script.textContent = '';
+    }
+})(\`${encodeURIComponent(codeRunningInPage)}\`);`;
+
+    browser.tabs.executeScript(tabId, {
+      code: codeRunningInContentScript,
+      runAt: 'document_start',
+      allFrames: true,
+      matchAboutBlank: true,
```

Comment:
there is no need to match aboutBlank, or is there?

Reply:
I think, it is needed (both allFrames and matchAboutBlank), since all iframes of the page have to be reached: https://stackoverflow.com/a/41410165/783510

Remi might confirm (or debunk), but I think matchAboutBlank was mainly introduced for the adblockers. Would need to test again, but I think there was breakage if it is left out.

Reply:
Yes, that is also my understanding. To match those, that option was introduced.